### PR TITLE
PP-6363 - Send address city to ePDQ 3DS2

### PR DESCRIPTION
**Pull request by nimalank7**
Description:
- If the address and city is present then send the parameter to ePDQ 3DS2

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2Order.java [hunk 1]
```diff
@@ -24,6 +24,7 @@
     public final static String BROWSER_ACCEPT_HEADER = "browserAcceptHeader";
     public final static String BROWSER_USER_AGENT = "browserUserAgent";
     public final static String BROWSER_JAVA_ENABLED = "browserJavaEnabled";
+    public final static String ADDRESS_CITY = "ECOM_BILLTO_POSTAL_CITY";
```
Comment (alexbishop1) [inline src/main/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2Order.java]: I think the constant should probably just match the `ECOM_BILLTO_POSTAL_CITY` value here, particularly as we’ll be sending the city in another field named `OWNERTOWN` as well.

File: src/main/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2Order.java [hunk 2]
```diff
@@ -53,6 +54,9 @@ public EpdqPayloadDefinitionForNew3ds2Order(String frontendUrl, SupportedLanguag
                 .add(BROWSER_ACCEPT_HEADER, getBrowserAcceptHeader(templateData))
                 .add(BROWSER_USER_AGENT, getBrowserUserAgent(templateData))
                 .add(BROWSER_JAVA_ENABLED, "false");
+        
+        templateData.getAuthCardDetails().getAddress().map(address -> address.getCity()).ifPresent(city -> parameterBuilder.add(ADDRESS_CITY, city));
```
Comment (alexbishop1) [inline src/main/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2Order.java]: You should be able to use a method reference in the first `map(…)`:

```java
templateData.getAuthCardDetails().getAddress().map(Address::getCity).ifPresent(city -> parameterBuilder.add(ADDRESS_CITY, city));
```

**Review by alexbishop1 — CHANGES_REQUESTED** [review]
Looks good: just one thing about the constant name.

File: src/main/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2Order.java (hunk 1 above)
Comment (nimalank7) [reply src/main/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2Order.java]: Done

File: src/main/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2Order.java (hunk 2 above)
Comment (nimalank7) [reply src/main/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2Order.java]: Done

**Review by alexbishop1 — APPROVED** [review]

**Pull request merged**